readthedocs/api/v3/permissions.py

```diff
@@ -0,0 +1,28 @@
+from rest_framework.permissions import IsAuthenticated
+
+
+class PublicDetailPrivateListing(IsAuthenticated):
+
+    """
+    Permission class for our custom use case.
+
+    * Always give permission for a ``detail`` request
+    * Only give permission for ``listing`` request if user is admin of the project
+    * Allow access to ``/projects`` (user's projects listing)
+    """
+
+    def has_permission(self, request, view):
+        is_authenticated = super().has_permission(request, view)
```

Comment:
Is this correct? auth and permission are different to me, so we should be sure we about what it is.

Reply:
I moved the chunk of specific code for ProjectViewSet and this class now basically does what David mentioned here:

1. returns user's projects if admin (`detail_objects` method)
1. returns all projects the user can see (`listing_objects` method)

We may want to keep talking about this in another channel. I don't want to block the PR on this piece of auth since we may need a bigger refactor for that anyways.